src/seedu/addressbook/data/person/Address.java

```diff
@@ -7,26 +7,35 @@
  * Guarantees: immutable; is valid as declared in {@link #isValidAddress(String)}
  */
 public class Address {
-
     public static final String EXAMPLE = "123, some street";
     public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
     public static final String ADDRESS_VALIDATION_REGEX = ".+";
 
-    public final String value;
+    private String value;
+    private Block block;
+    private Street street;
+    private Unit unit;
+    private PostalCode postalCode;
     private boolean isPrivate;
 
     /**
      * Validates given address.
      *
      * @throws IllegalValueException if given address string is invalid.
      */
-    public Address(String address, boolean isPrivate) throws IllegalValueException {
-        String trimmedAddress = address.trim();
+    public Address(Block block, Street street, Unit unit, PostalCode postalCode, boolean isPrivate) 
+            throws IllegalValueException {
         this.isPrivate = isPrivate;
-        if (!isValidAddress(trimmedAddress)) {
+
+        this.block = block;
+        this.street = street;
+        this.unit = unit;
+        this.postalCode = postalCode;
+        this.value = block + " " + street + " " + unit + " " + postalCode;
```

Comment:
since we're storing the various parts, we should not duplicate and store the value